### ISSUE TITLE
CPU 100% in version 1.2.1 update (previous version 1.1.9 works fine)

### ISSUE THREAD

**Issue by bwinter** [thuehlinger/daemons#27]
I have a pretty bland config `./bin/application.rb`:

```rb
require 'rubygems'
require 'daemons'

pwd = File.dirname(File.expand_path(__FILE__))
file = pwd + '/app.rb'

options = {
  :app_name   => "application",
  :dir_mode   => :normal,
  :dir        => File.join(pwd, '../tmp/pids'),
  :multiple   => false,
  :backtrace  => true,
  :log_output => true,
  :log_dir    => File.join(pwd, '../log'),
  :monitor    => true,
  :ontop      => false,
}

Daemons.run('./bin/app.rb', options)
```

`./bin/app.rb` is a process that polls a queue.

When I run this: `bundle exec ruby bin/application.rb start`, the `application_monitor` process starts, then within seconds it starts consuming more and more CPU cycles (until the CPU has no more to give). 

Additional problem is that the daemon doesn't respond properly to `stop` signals. It fails to die, won't take down the processes it spawned and it spawns a new process (essentially leaking them). I can spam it w/ `stops` and it will eventually terminate, leaving all the spawned processes for me to kill by hand.

Ruby Version 2.1.1
Daemon Version 1.2.1

For now I am just going to stick with version 1.1.9 but I thought I would report this in hopes of finding a resolution that allows me to stick w/ a newer version.

**Comment by ledermann** [comment]
Same here with Ruby 2.1.5, so I reverted to daemons v1.1.9, too.

**Comment by ryanjohns** [comment]
I'm seeing the same issue on ruby 2.2.0 and reverted to daemons 1.1.9.

**Comment by thuehlinger** [comment]
I think I found the issue behind this. I put out a 1.2.2 pre-release with the bug fixed. The gem is not yet on rubygems.org, but only available on github: https://github.com/thuehlinger/daemons/releases/download/v1.2.2/daemons-1.2.2.gem
@bwinter, @ledermann, @ryanjohns Could you test and report back?

**Comment by ryanjohns** [comment]
@thuehlinger 
I can confirm that it is working as expected for me on ruby 2.2.0. I did my test against the master branch rather than the .gem you linked but it looks the same. I performed a number of tests and all the monitoring-related functionality is working well for me. Thanks so much!

**Comment by ledermann** [comment]
@thuehlinger Tested with Ruby 2.1.5 and the master branch. Low CPU usage, thank you!

**Comment by thuehlinger** [comment]
Thanks. I've just officially released the 1.2.2 gem.
